### PR TITLE
TDL-7622: Support for EU endpoint

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
TDL-7622: Support for EU endpoint

Finalized approach:
- The EU endpoint works fine for all streams except revenue and export streams.
   - Revenue stream API endpoint for EU region returns 400 bad requests(Timeout Error)
   - Export stream API endpoint for EU region returns 200 ‘terminated early’
- As per discussion decided that let the customer come with the issue that these streams are not working. Skip the streams in the circleci.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_mixpanel/__init__.py [hunk 1]
```diff
@@ -47,7 +47,14 @@ def main():
         LOGGER.warning(
             "WARNING: Setting start_date to 1 year ago, {}".format(start_date))
 
+    #Check support for EU endpoints
+    if parsed_args.config.get('eu_residency_server'):
```
Comment (karanpanchal-crest) [inline tap_mixpanel/__init__.py:51]: @prijendev Please update this condition as per https://github.com/singer-io/tap-xero/blob/master/tap_xero/streams.py#L122

File: tap_mixpanel/streams.py [hunk 2]
```diff
@@ -297,6 +297,13 @@ def sync(self, state: dict, catalog, config: dict):
         days_interval = int(config.get("date_window_size", "30"))
         attribution_window = int(config.get("attribution_window", "5"))
 
+        #Update url if eu_residency_server is selected
+        if config.get('eu_residency_server'):
```
Comment (karanpanchal-crest) [inline tap_mixpanel/streams.py:301]: @prijendev Please update this condition as per https://github.com/singer-io/tap-xero/blob/master/tap_xero/streams.py#L122

File: tests/tap_tester/base.py [hunk 3]
```diff
@@ -110,16 +117,25 @@ def get_start_date(self):
...
+            
         for cred in creds:
             credentials_dict[cred] = os.getenv(creds[cred])
 
         return credentials_dict
 
     def expected_streams(self):
         """A set of expected stream names"""
+        
+        #Skip `export` and `revenue` stream for EU recidency server
```
Comment (karanpanchal-crest) [inline tests/tap_tester/base.py:135]: @prijendev Add reason for skipping these streams

File: tap_mixpanel/__init__.py (hunk 1 above)
Comment (prijendev) [reply tap_mixpanel/__init__.py]: done

File: tap_mixpanel/streams.py (hunk 2 above)
Comment (prijendev) [reply tap_mixpanel/streams.py]: done

File: tests/tap_tester/base.py (hunk 3 above)
Comment (prijendev) [reply tests/tap_tester/base.py]: done

**Comment by prijendev** [comment]
Raised to master.

**Pull request closed without merging**